components/UserCard.jsx

```diff
@@ -0,0 +1,29 @@
+import { UserContext } from "../context/User"
+import { useContext, useState } from "react"
+
+const UserCard = ({username, name, img}) => {
+    const{user, setUser} = useContext(UserContext)
+    const[signIn, setSignIn] = useState(false)
+    
+    const handleSignIn = () => {
+        if(!signIn){
+            setUser({...user, username: username})
+        } else {
+            setUser({...user, username: ""})
+        }
+        setSignIn(!signIn)
+    }
+    
+return(
+    <div className="card" style={{ width: "200px", margin: "10px"}} >
+        <img src={img} className="card-img-top" alt={username} style={{ width: "100%", height: "auto" }} />
+        <div className="card-body">
+            <p className="card-title">{username}</p>
+            <p className="card-text">{name}</p>
+            <button className="btn btn-primary" onClick={handleSignIn}>{signIn ? "Sign out" : "Sign in" }</button>
+            {user.username === username && (<p> You are logged in as {username}!</p>)}
+        </div>
+  </div>
```

Comment:
Nice work, I would also suggest a link that when the user is signed in, it takes them back to the articles. Not complete;y necessary but its nice user experience 🙂